src/strtod.cpp

```diff
@@ -167,25 +154,32 @@ double pj_strtod( const char *nptr, char **endptr )
 {
 /* -------------------------------------------------------------------- */
 /*  We are implementing a simple method here: copy the input string     */
-/*  into the temporary buffer, replace the specified decimal delimiter  */
-/*  with the one, taken from locale settings and use standard strtod()  */
+/*  into the temporary buffer, replace the specified decimal delimiter (.)  */
+/*  with the one taken from locale settings (ex ',') and then use standard strtod()  */
 /*  on that buffer.                                                     */
 /* -------------------------------------------------------------------- */
-    double      dfValue;
-    int         nError;
-    char        szWorkBuffer[PJ_STRTOD_WORK_BUFFER_SIZE];
+    char *pszNumber = replace_point_by_locale_point( nptr, '.' );
+    if ( pszNumber )
+    {
+        char *pszNumberEnd;
+        double dfValue = strtod( pszNumber, &pszNumberEnd );
 
-    char*       pszNumber = replace_point_by_locale_point(nptr, '.', szWorkBuffer);
+		int nError = errno;
 
-    dfValue = strtod( pszNumber, endptr );
-    nError = errno;
+		if ( endptr )
+		{
+            ptrdiff_t offset = pszNumberEnd - pszNumber;
+            *endptr = const_cast<char*>( nptr + offset );
+        }
 
-    if ( endptr )
-        *endptr = (char *)nptr + (*endptr - pszNumber);
+		free( pszNumber );
 
-    if (pszNumber != (char*) nptr && pszNumber != szWorkBuffer )
-        free( pszNumber );
+		errno = nError;
 
-    errno = nError;
-    return dfValue;
+        return dfValue;
```

Comment:
indentation is odd. Please use 4-space per indentation level

Reply:
Fixed